### ISSUE TITLE
Bespoke unable to download S3 file

### ISSUE THREAD

**Issue by ranjitiyer** [bespoken/bst#383]
Hello,

First of all, thanks for putting together this library. This is a huge help for the community. I was trying to use the library for encoding but ran into this issue. I've made sure that the AWS credentials are correctly setup, so I'm fairly certain that won't be the issue here. You're help is greatly appreciated!

Here's the code:

```
var bst = require("bespoken-tools");
var encoder = new bst.BSTEncode({ bucket: "S3Bucket" });
encoder.encodeURLAndPublishAs(
	"https://s3.amazonaws.com/path/to/file.mp3",
	"path/to/encoded-bespoken.mp3",
	(error, encodedURL) => {
		console.log("Error ? ", error)	
		console.log("Encoded Url ? ", encodedURL)	}
	);
```

Here's the output:
```
=> node bespoke-test.js 
Error ?  Error: Unable to download and save file at path https://s3.amazonaws.com/librivox-audio-books/sample.mp3. Error: Could not retrieve file from https://s3.amazonaws.com/librivox-audio-books/sample.mp3
    at ClientRequest.<anonymous> (/Users/ranjiti/work/node-test/aws-test/node_modules/bespoken-tools/lib/client/bst-encode.js:82:26)
    at ClientRequest.g (events.js:291:16)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:474:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
    at Socket.socketOnData (_http_client.js:363:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
Encoded Url ?  null
```

**Comment by jkelvie** [comment]
Hi @ranjitiyer thank you for the nice comments.

With regard to the error, the bucket you are uploading to is not publically accessible. 

An easy way to test this is to go into incognito mode and test the link in your browser.

FYI, the full list of requirements for audio files in SSML are here:
https://developer.amazon.com/public/solutions/alexa/alexa-skills-kit/docs/speech-synthesis-markup-language-ssml-reference#audio

We cover encoding it, but the bucket you will need to manage yourself. And please be careful about what you make accessible - I definitely recommend separating any bucket used for public audio files from any other content you have.

**Comment by ranjitiyer** [comment]
hi @jkelvie thanks for your quick turnaround and for pointing out permissions as the potential cause. I completely forgot that yours is an external service and would need permissions to read and write to the bucket.

So I went ahead and fixed permissions and gave 'Everyone' Read and Write access and that resolved the download issue. I'm now running into another error, presumably while attempting to write the encoded file back but I'm not able to figure out what the cause might be. Your guidance is highly appreciated. 

Here's the error. I've also pasted the source for you to look just in case I've provided the output S3 key incorrectly.

=> node bespoke-test.js 
Error ?  Error: All access to this object has been disabled
    at ClientRequest.<anonymous> (/Users/ranjiti/work/node-test/aws-test/node_modules/bespoken-tools/lib/client/bst-encode.js:82:26)
    at ClientRequest.g (events.js:291:16)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:474:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
    at Socket.socketOnData (_http_client.js:363:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
Encoded Url ?  null
=> 
=> 
=> 
=> more bespoke-test.js 
var bst = require("bespoken-tools");
var encoder = new bst.BSTEncode({ bucket: "S3Bucket" });
encoder.encodeURLAndPublishAs(
        "https://s3.amazonaws.com/librivox-audio-books/sample.mp3",
        "librivox-audio-books/sample-encoded-bespoken.mp3",
        (error, encodedURL) => {
                console.log("Error ? ", error)  
                console.log("Encoded Url ? ", encodedURL)       
        });

**Comment by jkelvie** [comment]
Hi, I believe the error you have is that the bucket should be set to `librivox-audio-books` in the encoder configuration (not S3Bucket) and the key (the second parameter to the encodeURLAndPublishAs call) should simply be `sample-encoded-bespoken.mp3`. Please let me know if that works for you!

**Comment by ranjitiyer** [comment]
Joy in the morning..It was a duh moment for me :). It works now!!

Just to give you some additional context about, I'm trying to build an Alexa audio book player for open source books. My source of books is librivox.org, an open source audio books initiative. They do a great job of making their books available on their web site and through an Android app. I want to make those great books available through the Echo.

Librivox doesn't have a centralized API to give me a dump of all their books and I've had to string together information from librivox.org and internet archive to build up a 'books database' that currently sits in ES. I'm going to be calling your API to encode Chapters on the fly as the user requests to play a book..

If this all works out, your API would save me considerable effort. I went down the path of standing up an AWS Lambda function that uses ffmpeg to do the encoding but I think due to a limitation in Lambda I couldn't fork off the ffmpeg process from the AWS Lambda infrastructure. I almost started whipping up a little NodeJS web service to do this but thankfully came across your API and heaved a big sigh of relief :)

Thanks for being so prompt in your responses. I may have more questions down the road as I get deeper with the integration :)
